sound/soc/sof/intel/hda-dsp.c

```diff
@@ -630,8 +630,8 @@ static int hda_suspend(struct snd_sof_dev *sdev, bool runtime_suspend)
 	snd_hdac_ext_bus_link_power_down_all(bus);
 #endif
 
-	/* power down DSP */
-	ret = snd_sof_dsp_core_power_down(sdev, chip->host_managed_cores_mask);
+	/* power down all DSP cores */
+	ret = snd_sof_dsp_core_power_down(sdev, GENMASK(chip->cores_num - 1, 0));
```

Comment:
No this doesn't look good to me.

On TGL, the driver cannot power up/down core1..3. Only Core0 can be powered up, and the other cores are powered using Core0 as a proxy through IPC. Setting the mask here is just not right.

And furthermore, if you look at the code of hda_dsp_core_reset_power_down(), there is a mask with no other than the host_managed_cores_mask, which on TGL is
````c
const struct sof_intel_dsp_desc tgl_chip_info = {
	/* Tigerlake , Alderlake */
	.cores_num = 4,
	.init_core_mask = 1,
	.host_managed_cores_mask = BIT(0),
````

````c
int hda_dsp_core_reset_power_down(struct snd_sof_dev *sdev,
				  unsigned int core_mask)
{
	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
	const struct sof_intel_dsp_desc *chip = hda->desc;
	int ret;

	/* restrict core_mask to host managed cores mask */
	core_mask &= chip->host_managed_cores_mask;
````

The question is thus: what does this patch actually do?

Reply:
@plbossart  Please see my clarfified commit message proposal. "host managed cores" is an Intel SOF DSP adaptation concept that SOF core does not know about. So if you use SOF core functions, you have to power down all cores and not apply the host managed cores mask.

Current code applies the mask and the state of cores1...3 on TGL will be not uptodate in SOF core and when a new pipelne is later sent requireding cores1...3, the IPC is not sent as SOF core (incorrectly) thinks cores1...3 are already enabled.

So I'd say this patch _is_ correct. It does reveal a conceptual issue in how we track core power state in SOF core, but I'd say that's a separate issue. Masking the cores this way in hda-dsp.c is not right.